server/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -212,7 +212,8 @@ public void writeBlobAtomic(final String blobName, final InputStream inputStream
     private void writeToPath(InputStream inputStream, Path tempBlobPath, long blobSize) throws IOException {
         try (OutputStream outputStream = Files.newOutputStream(tempBlobPath, StandardOpenOption.CREATE_NEW)) {
             final int bufferSize = blobStore.bufferSizeInBytes();
-            Streams.copy(inputStream, outputStream, new byte[blobSize < bufferSize ? Math.toIntExact(blobSize) : bufferSize]);
+            org.elasticsearch.core.internal.io.Streams.doCopy(
```

Comment:
I think we should consolidate the two `Streams` classes since the internal class was added for splitting the method out for a MR jar to take advantage of JDK9's `transferTo` method in #29322, but we don't use `transferTo` anymore for copying and on master we are already at JDK11 for a minimum.

Reply:
++ to that idea but maybe do it in another PR since that involves quite a few unrelated changes I think (since the one in `:server` uses `BytesReference` in a bunch of places and such).